### PR TITLE
Update User #105

### PR DESCRIPTION
**Pull request by Anastasiia2023**

### CODE REVIEW

**Review by TildaEndeavour — APPROVED** [review]

File: src/test/java/healthy/lifestyle/backend/users/controller/UserControllerTest.java [hunk 1]
```diff
@@ -95,10 +101,63 @@ void getAllCountriesTest_shouldReturnListOfCountriesAndStatusOk() throws Excepti
...
+            authorities = {"ROLE_USER"})
+    void updateUserTest_shouldReturnUpdatedUserAndStatusOk() throws Exception {
+        // Given
+        Role role = dataHelper.createRole("ROLE_USER");
+        Country country = dataHelper.createCountry(1);
+        Integer age = 20;
+        User user = dataHelper.createUser("one", role, country, null, age);
+        Country updatedCountry = dataHelper.createCountry(2);
+        UpdateUserRequestDto requestDto = dataHelper.createUpdateUserRequestDto("two", updatedCountry.getId(), 35);
+
```
Comment (OlegKDev) [inline src/test/java/healthy/lifestyle/backend/users/controller/UserControllerTest.java:124]: // When

File: src/test/java/healthy/lifestyle/backend/users/controller/UserControllerTest.java [hunk 2]
```diff
@@ -95,10 +101,63 @@ void getAllCountriesTest_shouldReturnListOfCountriesAndStatusOk() throws Excepti
...
+        mockMvc.perform(patch(UPDATE_USER_URL, user.getId())
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(requestDto)))
+                // Then
+                .andExpect(status().isOk())
```
Comment (OlegKDev) [inline src/test/java/healthy/lifestyle/backend/users/controller/UserControllerTest.java:129]: User id also `.andExpect(jsonPath("$.id", ...`

File: src/test/java/healthy/lifestyle/backend/users/controller/UserControllerTest.java [hunk 3]
```diff
@@ -95,10 +101,63 @@ void getAllCountriesTest_shouldReturnListOfCountriesAndStatusOk() throws Excepti
...
+    void updateUserTest_shouldReturn400_whenAuthenticationError() throws Exception {
+        // Given
+        Role role = dataHelper.createRole("ROLE_USER");
+        Country country = dataHelper.createCountry(1);
+        Integer age = 20;
+        User user = dataHelper.createUser("one", role, country, null, age);
+        User user2 = dataHelper.createUser("two", role, country, null, age);
+        Country updatedCountry = dataHelper.createCountry(2);
+        UpdateUserRequestDto requestDto = dataHelper.createUpdateUserRequestDto("two", updatedCountry.getId(), 35);
+
```
Comment (OlegKDev) [inline src/test/java/healthy/lifestyle/backend/users/controller/UserControllerTest.java:153]: // When

File: src/test/java/healthy/lifestyle/backend/users/service/UserServiceTest.java [hunk 4]
```diff
@@ -70,4 +76,33 @@ void createUserTest_shouldReturnUserDto() {
...
+                .fullName("oldFullName")
+                .age(35)
+                .country(country)
+                .build();
+
+        // When
+        when(userRepository.getReferenceById(userId)).thenReturn(existingUser);
+        when(countryRepository.getReferenceById(updateUserRequestDto.getUpdatedCountryId()))
+                .thenReturn(country);
+        when(userRepository.save(existingUser)).thenReturn(existingUser);
```
Comment (OlegKDev) [inline src/test/java/healthy/lifestyle/backend/users/service/UserServiceTest.java:97]: when(...) relates to the initial setup of a test, so it's something given before the test. In When section there is a method under a test, so "When" in a given-when-then relates to the main test event.

**Review by OlegKDev — CHANGES_REQUESTED** [review]

**Review by OlegKDev — APPROVED** [review]

**Pull request merged**